src/validation.cpp

```diff
@@ -1125,6 +1125,47 @@ bool ReadBlockFromDisk(CBlock& block, const CBlockIndex* pindex, const Consensus
     return true;
 }
 
+bool ReadRawBlockFromDisk(std::vector<uint8_t>& block, const CDiskBlockPos& pos, const CMessageHeader::MessageStartChars& messageStart)
+{
+    // Open history file to read
+    CDiskBlockPos hpos = pos;
+    hpos.nPos -= 8; // Seek back 8 bytes to validate header
+    CAutoFile filein(OpenBlockFile(hpos, true), SER_DISK, CLIENT_VERSION);
+    if (filein.IsNull()) {
+        return error("%s: OpenBlockFile failed for %s", __func__, pos.ToString());
+    }
+
+    CMessageHeader::MessageStartChars msg_start_in;
+    unsigned int nSize;
+    filein >> msg_start_in >> nSize;
+
+    if (memcmp(msg_start_in, messageStart, CMessageHeader::MESSAGE_START_SIZE)) {
+        return error("%s: Block magic mismatch for %s: %s versus expected %s", __func__, pos.ToString(),
+                HexStr(msg_start_in, msg_start_in + CMessageHeader::MESSAGE_START_SIZE),
+                HexStr(messageStart, messageStart + CMessageHeader::MESSAGE_START_SIZE));
+    }
+
+    try {
+        block.resize(nSize);
```

Comment:
Probably want to check the size is sane before we do this.

Reply:
Good point. What constant would be appropriate here?

Reply:
Another thing I wondered here: what is the C++11 proper way to allocate a vector (or a RAII memory area) without zeroing it?